src/dlnodesession.cpp

```diff
@@ -160,15 +160,23 @@ Status DLNodeSession::validate(const std::shared_ptr<ov::runtime::Tensor>& tenso
 
     // If batch size differs, check if remaining dimensions are equal
     const auto& dims = tensor->get_shape();
-    if (!tensorInfo.getShape()[0].match(dims[0])) {
+    const auto batchIndex = tensorInfo.getLayout().getBatchIndex();
+    if (!batchIndex.has_value() || tensorInfo.getShape().size() < batchIndex.value() + 1 || dims.size() < batchIndex.value() + 1) {
```

Comment:
MInor and could be skipped - wouldn't this be clearer?
```suggestion
    if (!batchIndex.has_value() ||
         batchIndex.value() >= tensorInfo.getShape().size() ||
         batchIndex.value() >=dims.size()) {
```
And anyway why we need to check if batchIndex.value() is not wrong within TensorInfo? At this point I think this shall not be needed - it may suggest that we missed something earlier on?

Reply:
TODO: Remove shape size vs dims size batch size index. To be done in separate task in next sprint (todo: update task description)

Reply:
done